src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandleExtensions.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Provides extension methods to operate with GC handles.
+    /// </summary>
+    public static class GCHandleExtensions
+    {
+        // The following methods are strongly typed generic specifications only on
+        // PinnedGCHandles with correct type.
+
+        /// <summary>
+        /// Retrieves the address of array data in <paramref name="handle"/>.
+        /// </summary>
+        /// <param name="handle">The handle to retrieve pointer from.</param>
+        /// <returns>
+        /// The address of 0th array element the pinned array,
+        /// or <see langword="null"/> if the handle doesn't point to any object.
+        /// </returns>
+        /// <exception cref="NullReferenceException">If the handle is not initialized or already disposed.</exception>
+        [CLSCompliant(false)]
+        public static unsafe T* GetAddressOfArrayData<T>(
+#nullable disable // Nullable oblivious because no covariance between PinnedGCHandle<T> and PinnedGCHandle<T?>
```

Comment:
@stephentoub Is `#nullable disable` correct here?

Reply:
That will make the parameter nullable oblivious.

Reply:
Yeah that was the intent here. When I suggested making the `T` nullable in API review I assumed that'd have allowed also passing a non nullable `T` just fine, but apparently that also warns, because reasons. So we figured just making this nullable oblivious would fix that. And keeping that restricted to the parameter means at least you preserve and flow the nullability to the returned `T`, rather than making the whole method nullable oblivious.